src/transformers/models/deta/modeling_deta.py

```diff
@@ -2204,10 +2220,13 @@ def forward(self, outputs, targets):
         num_boxes = sum(len(t["class_labels"]) for t in targets)
         num_boxes = torch.as_tensor([num_boxes], dtype=torch.float, device=next(iter(outputs.values())).device)
         # (Niels): comment out function below, distributed training to be added
-        # if is_dist_avail_and_initialized():
-        #     torch.distributed.all_reduce(num_boxes)
+        if dist.is_available() and dist.is_initialized():
+            torch.distributed.all_reduce(num_boxes)
+            world_size = dist.get_world_size()
+        else:
+            world_size = 1
```

Comment:
@amyeroberts I get that, however this will directly related to the normalization of loss function. Can you recommend other place to I implement this fucntion?

Reply:
cc @muellerzr, so sometimes when porting models (like DETA in this case), the modeling code includes some `torch.distributed` utilities like here, the number of bounding boxes needs to be divided by `world_size`.

Wondering what your recommendation is regarding making this code compatible with frameworks like Accelerate/Trainer API.

Reply:
The safe way is to use Accelerate's reduce method instead and is my recommendation

```python
from accelerate import PartialState
from accelerate.utils import reduce

# Check that we have initialized the distributed state
world_size = 1
if PartialState._shared_state != {}:
    num_boxes = reduce(num_boxes)
    world_size = PartialState().num_processes
```

Reply:
Opened an issue here: https://github.com/huggingface/transformers/issues/28309

Reply:
Even though the loss graph has dramatically changed at the initial part, it seems okay with the custom dataset!

Reply:
@NielsRogge @muellerzr Thanks for the recommendation. Suggested code seems to work properly and changed with the last commit.